### ISSUE TITLE
Profiles should support multiple email addresses

### ISSUE THREAD

**Issue by jrk** [keybase/keybase-issues#105]
Most of us have multiple email addresses which should be associated with a single cryptographic identity.

**Comment by phillipadsmith** [comment]
@stumyp Super, thanks! :)

(Did you edit your key with the GPG Tools UI or a command-line tool?)

**Comment by stumyp** [comment]
@phillipadsmith 
I used GPG UI to upload my identities and other fields in the key
and  keybase CLI tool to push it back to keybase.io

**Comment by phillipadsmith** [comment]
@stumyp Thanks! :)

**Comment by ZacharyDuBois** [comment]
So does this work using GPG Tools on Mac? :) I'd love to see multiple email support.

**Comment by stumyp** [comment]
I did it all on Mac GPG Tools. I bet it will work on Linux or Windows no problem too

**Comment by ZacharyDuBois** [comment]
Sweet. Keybase should add this for normal use.

**Comment by vrillusions** [comment]
:+1: 
My use case is similar to others here.  I have a gpg key, [keybase profile](https://keybase.io/vrillusions) and it has two email addresses.  One is what I've used for like 15 years.  The other is a new one that I added a year or so ago.  They both go to me and can be used interchangeably, mainly because I'll never get rid of the older one. 

The reason I came searching for this issue to report it because I just got a keybase invite to that new address.  Well I'm going to tell them I already have one at this other address.  The problem is going to be I'll tell him that and then I get another invite from someone going "hey this guy isn't on here I'm going to send an invite".  To put it another way, there should be a way to say on keybase that if anyone searches for these certain emails that it's this account.  The automagical way would be to check all the ids in the gpg key and just mark all those as aliases of this profile.

**Comment by henrylawson** [comment]
+1 for the feature or some advice on how to handle this.

Ideally I'd like to associate multiple emails with my keybase account that are searchable to reveal my keybase ID however not listed publicly on my profile.

**Comment by fearoffish** [comment]
I've got multiple email addresses in my key, which I push up with `keybase push -u`. However when others get my pub key it doesn't show them multiple emails. What am I missing from what you all did?

**Comment by fearoffish** [comment]
Side note, running: `curl https://keybase.io/USER/key.asc | gpg --import` updates the local key which I thought re-tracking did. I tried deleting the local gpg key and untracking but using the keybase cli still pulled the old one.

**Comment by stumyp** [comment]
@fearoffish to get updated key keybase users should do untrack/track. 
hope it can be fixed soon ...

**Comment by jsatk** [comment]
@stumyp I'm in the same boat as @fearoffish. I untracked and re-tracked but it pulls the old one. If I run the command @fearoffish pasted above it pulls in the correct one.

**Comment by pts** [comment]
+1

**Comment by melalj** [comment]
+1

**Comment by mamayer** [comment]
+1

**Comment by jacoscaz** [comment]
+1

**Comment by lavalamp-** [comment]
+1

**Comment by mholt** [comment]
@zQueal 
> Quite honestly, I'd be much happier with a them@keybase.io mail relay. By sending email to that address a user can be ensured that they're sending it to the verified email address on the account. It would help those who might not want to display their email address publicly.

But [emails to keybase.io blackhole](https://github.com/keybase/keybase-issues/issues/941#issuecomment-185218865). Please let us add a working email address to the key, as I've had to export and modify my key to work with signing GitHub commits now.

**Comment by maxtaco** [comment]
BTW, for those making new keys with the Web client or Go client, we now allow multiple emails of your choosing, and of course don't include the `foo@keybase.io` email address anymore. We just haven't built out tooling to modify old keys.

And just in case everyone didn't already know this, when you change your UIDs in PGP, you don't change your key fingerprint.  So you wouldn't have to retrack anyone or have anyone retrack you. That's just the way PGP works!

**Comment by mholt** [comment]
Nice, thanks for the update @maxtaco -- would be great to have that rolled out to us early adopters too. :smile:

**Comment by maxtaco** [comment]
Yeah, for you guys, the current advice sadly is to use `gpg --edit-key`. More elbow grease required... for now...

**Comment by jgrevich** [comment]
@maxtaco I removed the keybase email from the key using `gpg --edit-key` however when I try to update the key to the keybase.io server I get the following error with the keybase client

```
▶ ERROR You can't delete your keybase.io username (error 915)
```

And a similar error with the curl command:

```
Error in your post
-------------------
Code: 915
Name: KEY_BAD_UIDS
Description: You can't delete your keybase.io username
```

Did I do something wrong?  I tried adding a new UID with `keybase.io/jgrevich` as my real name and a blank email but that results in the same error. It seems like this is related to https://github.com/keybase/keybase-issues/issues/1911. Is that still being worked on? 

Is there a way for me to update this key without including the keybase.io email so I can also use it on github.com. Alternatively, is there a way to setup a mail relay for the keybase.io account?

**Comment by BretFisher** [comment]
How I got this to work so I can use keybase GPG with github.com on Mac:

 - Use keybase to create a pgp if you don't have one
 - Using keybase cli: selected my proper key: `keybase pgp select`
 - Used [GPG Keychain](https://gpgtools.org/) GUI to add all my email addresses (would love to know the proper step by step using `gpg --edit-key "keybase.io/username"` but didn't take the time)
 - Used keybase cli to update public `keybase pgp update`
 - Copy public key to clipboard: `keybase pgp export | pbcopy`
 - Paste into github settings: https://github.com/settings/keys (it will show all emails you added)
 - Update git locally to sign commits: https://github.com/pstadler/keybase-gpg-github

**Comment by compumike08** [comment]
I think this issue is the same as #2266.

**Comment by nmaas87** [comment]
+1

**Comment by nickik** [comment]
What if I don't want to use GPG anymore? I would like to have E-Mail associated with Keybase, so that people (and hopefully) E-Mail clients can send me Mail that is encrypted with Saltpack?

**Comment by ulope** [comment]
gpg steps for @BretFisher's howto:

1. `gpg --edit-key "keybase.io/username"`
2. `uid 1` - select the existing `@keybase.io` "email" (called uid in gpg lingo)
3. `deluid` - delete it
4. `adduid` - enter details of new email
5. `quit` - save changes: yes
